network/src/peers/peer/connector.rs

```diff
@@ -0,0 +1,93 @@
+// Copyright (C) 2019-2021 Aleo Systems Inc.
+// This file is part of the snarkOS library.
+
+// The snarkOS library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The snarkOS library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with the snarkOS library. If not, see <https://www.gnu.org/licenses/>.
+
+use std::{
+    io::{Error as IoError, ErrorKind},
+    time::Duration,
+};
+
+use futures::{select, FutureExt};
+use snarkvm_objects::Storage;
+use tokio::{net::TcpStream, sync::mpsc};
+
+use super::PeerAction;
+use crate::{stats, NetworkError, Node, Peer, PeerEvent, PeerEventData, PeerHandle, Version};
+
+use super::network::PeerNetwork;
+
+const CONNECTION_TIMEOUT_SECS: u64 = 3;
```

Comment:
we might want to keep it with the other crate-level consts for easier tweaking, and I think we already have one for it there

Reply:
is it a crate-level const? it's only used for connections, it make sense to keep it local to the connector to me. open to further discussion.

Reply:
This should be moved to where all of the other magical constants are set